cmd/clusterctl/pkg/client/cluster/mover.go

```diff
@@ -229,6 +236,67 @@ func setClusterPause(proxy Proxy, clusters []*node, value bool, log logr.Logger)
 	return nil
 }
 
+// ensureNamespaces ensures all the expected target namespaces are in place before creating objects.
+func (o *objectMover) ensureNamespaces(graph *objectGraph, toProxy Proxy) error {
+	cs, err := toProxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	namespaces := sets.NewString()
+	for _, node := range graph.getNodesWithClusterTenants() {
+		namespace := node.identity.Namespace
+
+		// If the namespace was already processed, skip it.
+		if namespaces.Has(namespace) {
+			continue
+		}
+		namespaces.Insert(namespace)
+
+		// Otherwise check if namespace exists (also dealing with RBAC restrictions).
+		ns := &corev1.Namespace{}
+		key := client.ObjectKey{
+			Name: namespace,
+		}
+
+		if err := cs.Get(ctx, key, ns); err == nil {
+			return nil
+		}
+		if apierrors.IsForbidden(err) {
+			namespaces := &corev1.NamespaceList{}
+			if err := cs.List(ctx, namespaces); err != nil {
+				return err
+			}
+
+			for _, ns := range namespaces.Items {
+				if ns.Name == namespace {
+					return nil
+				}
+			}
+		}
```

Comment:
Is there a particular use case that we're dealing with here? I'm wondering why this fallback to list is necessary. 

Also shouldn't we check the `Continue` ID?

Reply:
This is making the code tolerant to use cases where the user is allowed to list namespaces but not to get namespaces (copied from the old clusterctl, and it makes sense IMO)

> Also shouldn't we check the Continue ID?
Fixed

this is not fully clear to me...